Course/Course/Data/Model/Data_CourseOutlineResponse.swift

```diff
@@ -64,7 +90,44 @@ public extension DataLayer {
             certificate = try values.decode(Certificate.self, forKey: .certificate)
             org = try values.decode(String.self, forKey: .org)
             isSelfPaced = try values.decode(Bool.self, forKey: .isSelfPaced)
+            courseModes = try values.decode([CourseMode].self, forKey: .courseModes)
```

Comment:
If these properties are optional, `try?` should be used here. Otherwise a error will occur if they are omitted in the data. WDYT?